glusterd2/commands/volumes/volfiles.go

```diff
@@ -33,3 +37,38 @@ func volfilesListHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, volfiles)
 }
+
+func volfilesGetHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	volfiles, err := volgen.GetVolfiles()
+
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "unable to get list of volfiles")
+		return
+	}
+
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, volfiles)
+}
+
+func volfileGetHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	volfile := mux.Vars(r)["volfileid"]
+
+	//remove extra slash in last ( as StrictSlash is false in mux.Route)
+	volfile = strings.TrimSuffix(volfile, "/")
+	volfiles, err := volgen.GetVolfile(volfile)
+
+	if err != nil {
+		if err == errors.ErrVolFileNotFound {
+			restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolFileNotFound)
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, fmt.Sprintf("unable to fetch volfile content %s", volfile))
+		}
+		return
+	}
+
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
```

Comment:
not required, `SendHTTPResponse` is setting this

Reply:
if I pass nil as the response it won't set the header. need to do it here

Reply:
@aravindavk updated content type to text/plain